### PR TITLE
Adding TFT API support

### PR DESCRIPTION
**Pull request by happsie**
Added support for the following TFT APIs:

1. SPECTATOR-TFT-V5
2. TFT-LEAGUE-V1
3. TFT-MATCH-V1
4. TFT-STATUS-V1
5. TFT-SUMMONER-V1

Had to make a minor change to the `client.go` adding request options. This is because the `SUMMONER-V1` has en endpoint that requires setting an authorization header. So, instead of breaking all methods, I thought adding a functional options pattern was better. This would also allow for other extensions of the request in the future. 

Adding this as a draft since this is not related to an active issue and is rather big. I will continue the godoc of the `models.go`

Also found out that the API is a bit out of sync for what it seems. Running live API-calls would sometime give other properties than specified in the API. I have added some of the missing fields and removed some to try match the API as close as possibly without making it hard to understand.

### CODE REVIEW

File: internal/request_options.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import (
+	"fmt"
+	"net/http"
+)
+
+type RequestOption func(r *http.Request)
+
+func WithHeader(key, value string) RequestOption {
+	return func(r *http.Request) {
+		fmt.Println(value)
```
Comment (happsie) [inline internal/request_options.go:12]: Remove print statement

**Comment by happsie** [comment]
@KnutZuidema feel free to have a look when you have time. Not sure if you got notification about this

**Comment by happsie** [comment]
Any updates on this @KnutZuidema ? 😄

File: riot/tft/summoner.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		return nil, err
+	}
+	return out, nil
+}
+
+// GetSummonerByMe returns a summoner by access token
+func (sc *SummonerClient) GetSummonerByMe(authorization string) (*Summoner, error) {
+	logger := sc.logger().WithField("method", "GetSummonerByMe")
+	var out *Summoner
+	if err := sc.c.GetInto(endpointSummonerByMe, &out, internal.WithHeader("Authorization", authorization)); err != nil {
```
Comment (KnutZuidema) [inline riot/tft/summoner.go:42]: I don't understand the need for this `WithHeader` option. The api key is already available to the client, why do we need additional authorization? Is it different from the API key?

File: riot/tft/constants.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
+package tft
+
+const (
+	endpointBase          = "/tft"
+	endpointSpectatorBase = "/lol/spectator" + endpointBase
```
Comment (KnutZuidema) [inline riot/tft/constants.go:5]: is this intended? This uses the `/lol` base path.

**Review by KnutZuidema — COMMENTED** [review]
thanks for the effort @happsie

File: riot/tft/constants.go (hunk 3 above)
Comment (happsie) [reply riot/tft/constants.go:5]: Yes, actually, it uses /lol/ as base path for the spectator. https://developer.riotgames.com/apis#spectator-tft-v5

File: riot/tft/summoner.go (hunk 2 above)
Comment (happsie) [reply riot/tft/summoner.go:42]: This is the API: https://developer.riotgames.com/apis#tft-summoner-v1/GET_getByAccessToken

I had a look at it, and it seem a bit confusing. So golio passes the header `X-Riot-Token` as a header for authenticating the application. But this API requires another token called `Authorization`, it's not specified what this authorization is. But looking at the menu subtitle may give a hint of what it is. The title say `Teamfight tactics RSO`, so the Authorization header that is optional in this API may be the `Riot Sign on`-token. What do you think @KnutZuidema

**Comment by happsie** [comment]
> thanks for the effort @happsie

No problem, anything more that needs fixing?

**Review by KnutZuidema — APPROVED** [review]

**Comment by happsie** [comment]
> Thanks @happsie 🙏🏼

Thank you for making this awesome project!

**Pull request merged**